### PR TITLE
Task/14280 add entry to contact journal

### PR DESCRIPTION
**Pull request by 30mar**
## Description
- Add entry to contact Journal after successful submission
- Handel database migration

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-14280

## Screenshots
Screenshot 2022-12-07 at 16 09 41

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayTableViewCell.swift [hunk 1]
```diff
@@ -195,6 +199,72 @@ class DiaryOverviewDayTableViewCell: UITableViewCell {
...
+		// submissions
+		submissionsStackView.arrangedSubviews.forEach { $0.removeFromSuperview() }
+		submissionsStackView.isHidden = cellViewModel.diaryDaySubmissions.isEmpty
+		if #available(iOS 14.0, *) {
+			submissionsStackView.backgroundColor = .enaColor(for: .darkBackground)
+		} else {
+			submissionsStackView.add(backgroundColor: .enaColor(for: .darkBackground))
+		}
+		// Because we set the background color, the border of the underlying view disappears. For this we need some new borders at the left and right.
+		drawBorders(to: [.left, .right], on: submissionsStackView)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Overview/Cells/DiaryOverviewDayTableViewCell.swift:213]: 👍

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift [hunk 2]
```diff
@@ -221,13 +221,21 @@ class ContactDiaryStore: DiaryStoring, DiaryProviding, SecureSQLStore {
 				DELETE FROM CoronaTest
 				WHERE date < date('\(todayDateString)','-\(dataRetentionPeriodInDays - 1) days')
 			"""
+			
+			let sqlCoronaSubmissions = """
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift:225]: 👍

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift [hunk 3]
```diff
@@ -549,6 +557,44 @@ class ContactDiaryStore: DiaryStoring, DiaryProviding, SecureSQLStore {
 
 		return _result
 	}
+	
+	func addSubmission(date: String) -> SecureSQLStore.IdResult {
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift:561]: Better to call it keySubmission ?

```suggestion
	func addKeySubmission(date: String) -> SecureSQLStore.IdResult {
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift [hunk 4]
```diff
@@ -549,6 +557,44 @@ class ContactDiaryStore: DiaryStoring, DiaryProviding, SecureSQLStore {
...
+		var result: SecureSQLStore.IdResult?
+
+		databaseQueue.inDatabase { database in
+			Log.info("[ContactDiaryStore] Add Submission.", log: .localData)
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/ContactDiaryStore.swift:565]:
```suggestion
			Log.info("[ContactDiaryStore] Add Key Submission.", log: .localData)
```

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/StoreV6/ContactDiaryMigration5To6.swift [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+//
+
+import FMDB
+
+final class ContactDiaryMigration5To6: Migration {
+
+	private let databaseQueue: FMDatabaseQueue
+	private var error: Error?
+
+	init(databaseQueue: FMDatabaseQueue) {
```
Comment (flxschmidt) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/StoreV6/ContactDiaryMigration5To6.swift:12]: Private down and Mark, Init Mark

**Review by flxschmidt — APPROVED** [review]
Great Job!

File: src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/MockDiaryStore.swift [hunk 6]
```diff
@@ -77,6 +77,13 @@ class MockDiaryStore: DiaryStoringProviding {
 		coronaTests.append(diaryDayTest)
 		return .success(id)
 	}
+	
+	@discardableResult
+	func addSubmission(date: String) -> Result<Int, SecureSQLStoreError> {
+		let id = (submissions.map { $0.id }.max() ?? -1) + 1
+		submissions.append(DiaryDaySubmission(id: id, date: date))
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/ContactDiary/Store/__tests__/MockDiaryStore.swift:84]: 👍

**Review by naveeddotio — APPROVED** [review]

**Comment by naveeddotio** [comment]
Looks nice to me, good work 👍

**Comment by Ein-Tim** [comment]
Can we get a screenshot of the change, or is the UI not yet finished?

**Review by puneetmahali1 — APPROVED** [review]

**Comment by naveeddotio** [comment]
> Can we get a screenshot of the change, or is the UI not yet finished?

@Ein-Tim Video is added for the reference.

**Pull request merged**